src/Form/Type/ChoiceFieldMaskType.php

```diff
@@ -33,8 +33,11 @@ public function buildView(FormView $view, FormInterface $form, array $options)
             }
         }
 
-        $allFieldNames = call_user_func_array('array_merge', $sanitizedMap);
-        $allFieldNames = array_unique($allFieldNames);
+        $allFieldNames = [];
+        if ([] !== $sanitizedMap) {
```

Comment:
why is this if needed?

Reply:
If `$santizedMap` is an empty array, `call_user_func_array('array_merge', $sanitizedMap)` will throw a `Warning: array_merge() expects at least 1 parameter, 0 given`.

Reply:
The only problem here is even if the array is not empty, if the value of each item is not an array it will still fail.

Reply:
Question, what code is easier to understand (I don't have an opinion yet):

```
        $allFieldNames = [];
        if ([] !== $sanitizedMap) {
            $allFieldNames = call_user_func_array('array_merge', $sanitizedMap);
            $allFieldNames = array_unique($allFieldNames);
        }
```

or:

```
$allFieldNames = array_unique(array_reduce($sanitizedMap, function($carry, $value) {
    return array_merge($carry, is_array($value) ? $value : []);
}, []));
```

or:

```
$allFieldNames = [];
foreach ($sanitizedMap as $value) {
    if (is_array($value)) {
        $allFieldNames = array_merge($allFieldNames, $value);
    }
}
$allFieldNames = array_unique($allFieldNames);
```

Reply:
How about `$allFieldNames = array_merge(...$sanitizedMap);`?

Reply:
I think the last one is easiest to understand. I do like the version @greg0ire suggested. I'll run some tests to see if it'll work in all cases.

Reply:
it fails on my test: https://3v4l.org/sDsvL

Reply:
Ah right, needs array values https://3v4l.org/b6sdI

Reply:
I'll also set up a unit test to test weird values in the second layer of the `$map` array.

Reply:
Thank you both for your feedback and insights. I changed the code and added a unit test to test for unexpected values.